services/repository/repository.go

```diff
@@ -32,6 +31,11 @@ func CreateRepository(doer, owner *models.User, opts models.CreateRepoOptions) (
 func ForkRepository(doer, u *models.User, oldRepo *models.Repository, name, desc string) (*models.Repository, error) {
 	repo, err := models.ForkRepository(doer, u, oldRepo, name, desc)
 	if err != nil {
+		if repo != nil {
```

Comment:
The problem is that all the exit paths from `models.ForkRepository()` return `nil, err`, so this condition will never be met.

Reply:
done.

Reply:
The git operations will not be reverted if -for example- the first commit in `models.ForkRepository()` fails. Is that OK?

Reply:
@guillep2k I also think `ForkRepository`'s logic has some problem and I think I can send another PR to fix it since this PR is just a moving refactor.